crates/client/src/client.rs

```diff
@@ -247,8 +247,13 @@ impl FuelClient {
                 })?;
         }
 
-        let client =
-            client_builder.build_with_conn(es::HttpsConnector::with_webpki_roots());
+        let client = client_builder.build_with_conn(
+            hyper_rustls::HttpsConnectorBuilder::new()
+                .with_webpki_roots()
+                .https_only()
```

Comment:
Why do we want to use only `https`? I think it is better to allow all [`https_or_http`](https://docs.rs/hyper-rustls/latest/hyper_rustls/struct.HttpsConnectorBuilder.html#method.https_or_http)

It is useful when you need to connect locally via `127.0.0.1`